### DISCUSSION TITLE
Is the API_KEY generated always unique?

### DISCUSSION THREAD

**Discussion by hilmanski** [florimondmanca/djangorestframework-api-key#239] [Q&A]
Hi

since this key will be used for different users.
Is the API_KEY generated always unique? does it have sort of "database" to remember what it has used to make sure no duplication ?

**Comment by florimondmanca** [comment]
Hi @hilmanski,

Unless I misremember, there is no such mechanism in place for the entire API key. Only the prefix is marked with a UNIQUE constraint. But I don’t think we check ahead of time or retry should the same prefix be picked. The package relies on the low collision probability of long strings of tokens for now.

I’ll move this to a discussion but if there’s a specific problem you’re encountering because of this, feel free to file a detailed ticket.
